client/dashboard/emails/add-professional-email/index.tsx

```diff
@@ -0,0 +1,303 @@
+import { useDispatch } from '@wordpress/data';
+import { __, sprintf } from '@wordpress/i18n';
+import { store as noticesStore } from '@wordpress/notices';
+import { add } from 'date-fns';
+import { useCallback, useEffect, useState } from 'react';
+import { useAuth } from '../../app/auth';
+import { useLocale } from '../../app/locale';
+import { ButtonStack } from '../../components/button-stack';
+import { PageHeader } from '../../components/page-header';
+import PageLayout from '../../components/page-layout';
+import { Text } from '../../components/text';
+import { CartActionError } from '../../shopping-cart/errors';
+import { formatDate } from '../../utils/datetime';
+import { getEmailCheckoutPath } from '../../utils/email-paths';
+import {
+	FIELD_NAME,
+	FIELD_FIRSTNAME,
+	FIELD_LASTNAME,
+	FIELD_IS_ADMIN,
+	FIELD_PASSWORD_RESET_EMAIL,
+	FIELD_MAILBOX,
+	FIELD_PASSWORD,
+} from '../entities/constants';
+import { MailboxForm as MailboxFormEntity } from '../entities/mailbox-form';
+import { MailboxOperations } from '../entities/mailbox-operations';
+import { FormFieldNames, MutableFormFieldNames, SupportedEmailProvider } from '../entities/types';
+import { IntervalLength } from '../types';
+import { getCartItems } from '../utils/get-cart-items';
+import { getEmailProductProperties } from '../utils/get-email-product-properties';
+import { getProductSlugForProviderAndInterval } from '../utils/get-product-slug-for-provider-and-interval';
+import { EmailNonDomainOwnerNotice } from './components/email-non-domain-owner-notice';
+import { MailboxForm } from './components/mailbox-form';
+
+type HiddenFieldNames = Exclude<
+	MutableFormFieldNames,
+	typeof FIELD_MAILBOX | typeof FIELD_PASSWORD
+>;
+
+const possibleHiddenFieldNames: HiddenFieldNames[] = [
+	FIELD_NAME,
+	FIELD_FIRSTNAME,
+	FIELD_LASTNAME,
+	FIELD_IS_ADMIN,
+	FIELD_PASSWORD_RESET_EMAIL,
+];
+
+const AddProfessionalEmail = () => {
+	const { user } = useAuth();
+	const { createErrorNotice } = useDispatch( noticesStore );
+	const locale = useLocale();
+	const router = useRouter();
+	// Extract params from the current match for this route
+	const match = router.state.matches[ router.state.matches.length - 1 ];
+	const params = ( match?.params ?? {} ) as { domain?: string; type?: string };
+	const { domain: domainName = '' } = params;
+
+	let interval: IntervalLength = router.state.location.search.interval;
+	if ( interval !== 'monthly' && interval !== 'annually' ) {
+		interval = 'annually';
+	}
+
+	const { data: domain, isFetched: isDomainFetched } = useQuery( domainQuery( domainName ) );
+	const userCanAddEmail = domain?.current_user_can_add_email;
+	const { data: products } = useQuery( productsQuery() );
+	const productSlug = getProductSlugForProviderAndInterval( 'titan', interval );
+	const product = products?.[ productSlug ] as Product;
+	const { data: site } = useQuery( siteBySlugQuery( domainName ) );
```

Comment:
One site can more than one domain, so you actually need to fetch the site attached to the domain

Reply:
Addressed on https://github.com/Automattic/wp-calypso/pull/106524/commits/527e2586d3200d84da88ef9742a1a8448f37c8de